HW20/Game.cpp

```diff
@@ -0,0 +1,313 @@
+		return UserInput::ApplyModifier;
+	default:
+		return UserInput::None;
+	}
+}
+
+int selectModifierFromHand(unsigned short handSize)
+{
+	std::cout << "Select card from HAND: ";
+	int choice = -1;
+	std::cin >> choice;
+
+	if (choice >= handSize || choice <= 0)
+	{
+		choice = -1;
+	}
+
+	return choice;
+}
+
+void printCurrentFightResult(int powerDifference)
+{
+	std::cout << "You are missing " << std::abs(powerDifference) << " power!\n\n";
+}
+
+void printMonsterInfo(Monster* monster)
+{
+	std::cout << "\n--------Monster \"" << monster->getName() << "\"" << ", of " <<
+		UI::getTribeString(monster->getTribe()) << ", level " << monster->getLevel() << " --------" << std::endl << std::endl;
+	//#TODO: Print RUNAWAY POLICIES info similar to items print in printPlayerDeck()
+}
+
+} //namespace UI
+
+
+void Game::run()
+{
+	constexpr uint WinningLevel = 10;
+	loadDataBase("MunchkinDataBase.txt");
+
+	std::srand(static_cast<int>(std::time(0)));
+	UI::printIntro(&m_munchkin);
+	generateMunchkinInitialCards();
+
+	while (m_munchkin.getLevel() < WinningLevel)
+	{
+		UI::printPlayerDeck(&m_munchkin);
+		
+		UI::pressAnyKeyToContinue();
+		UI::printMonsterSelection();
+
+		Monster* monster = generateMonster();
+		UI::printMonsterInfo(monster);
+
+		Fight fight;
+		fight.setMonster(monster);
+		fight.setMunchkin(&m_munchkin);
+
+		fight.start();	//Initialization, calculate monster power, munchkin
+
+		//State pattern may be a good candidate here
+		//Every case may be its own state with transition rules, e.g.
+		//Start->InProgress->Win/Runaway/ApplyModifiers, Runaway->Lost, ApplyModifiers->InProgress
+		//https://refactoring.guru/uk/design-patterns/state
+		
+		//Fight loop
+		while (!fight.getFinish())
+		{
+			const int munchkinMonsterDifference = fight.getMunchkinMonsterPowerDifference();
+			UI::printMunchkinPower(&m_munchkin, &fight);
+			UI::printMonsterPower(monster, &fight);
+
+			if (munchkinMonsterDifference > 0)	//WINNING CASE
+			{
+				fight.victoryFlow(m_itemsDeck, m_modifiersDeck);
+				UI::printMunchkinWon();
+				UI::pressAnyKeyToContinue();
+			}
+			else  //LOSING CASE
+			{
+				UI::printCurrentFightResult(munchkinMonsterDifference);
+				UI::UserInput input = UI::readUserBehavior();
+				if (input == UI::UserInput::Runaway)
+				{
+					fight.runawayFlow();
+
+					UI::printMunchkinLost(monster);
+					UI::pressAnyKeyToContinue();
+				}
+				else if (input == UI::UserInput::ApplyModifier)
+				{
+					UI::printModifiers(&m_munchkin);
+					const int choice = UI::selectModifierFromHand(static_cast<short>(m_munchkin.getModifiers().size()));
+					
+					if (choice != -1)
+						fight.applyModifier(choice - 1);
+				}
+			}
+		}
+	}
+}
+
+void Game::generateMunchkinInitialCards()
+{
+	m_munchkin.addItems(m_itemsDeck.generateItems(5));
+	m_munchkin.addModifiers(m_modifiersDeck.generateModifiers(10));
+}
+
+Monster* Game::generateMonster()
+{
+	return m_monstersDeck.getRandomCard();
+}
+
+void Game::loadDataBase(std::string dataBaseFile)
```

Comment:
better make it part of the Deck, or create another class, e.g. DeckDatabaseLoader, and call it in c-tor of the Deck class